### PR TITLE
Remove persistence from Spree::Order#associate_user!

### PR DESCRIPTION
**Pull request by dkubb**
This method should not be responsible for persisting the data change. It should only copy over the data to the order and it should be the responsibility of the controller or caller to persist the changed order.

### CODE REVIEW

**Comment by dkubb** [comment]
@mbj assuming green, this is ready for review.

**Comment by dkubb** [comment]
@mbj do you think I should mutation test some of this code I touched?

**Comment by dkubb** [comment]
This change was verified with:

```bash
RAILS_ENV=test bundle exec mutant \
  --require ./spec/dummy/config/environment \
  --jobs 1 \
  --use rspec \
  -- \
  'Spree::Order#associate_user'
```

**Comment by dkubb** [comment]
@mbj this is ready for review.

File: backend/app/controllers/spree/admin/orders/customer_details_controller.rb [hunk 1]
```diff
@@ -21,7 +21,7 @@ def edit
         def update
           if @order.update_attributes(order_params)
             if params[:guest_checkout] == "false"
-              @order.associate_user!(Spree.user_class.find(params[:user_id]), @order.email.blank?)
+              @order.associate_user(Spree.user_class.find(params[:user_id]), @order.email.blank?)
```
Comment (mbj) [inline backend/app/controllers/spree/admin/orders/customer_details_controller.rb]: As `Order#associate_user` assumes `user` is not nil, and `Model.find` will either return an object or raise an exception, I think that we can safely use the stronger for m `HashWithIndifferentAccess#fetch`, as passing `nil` to `.find` is not a valid use case?

**Comment by mbj** [comment]
@dkubb I finished a pass over this, apart from one flag I'm fine.

File: backend/app/controllers/spree/admin/orders/customer_details_controller.rb (hunk 1 above)
Comment (dkubb) [inline backend/app/controllers/spree/admin/orders/customer_details_controller.rb]: I agree. I will make this change.

**Comment by dkubb** [comment]
@mbj this is ready for another pass.

**Comment by mbj** [comment]
@dkubb I'm fine with this to merge. After we deployed the current master HEAD given our 1:1 policy.

**Comment by dkubb** [comment]
> After we deployed the current master HEAD given our 1:1 policy.

Yup, I'll do this as soon as it's merged. I have a local branch that is ready to push.

**Comment by mbj** [comment]
I just pushed one.

**Pull request merged**